src/core/server/http/integration_tests/lifecycle.test.ts

```diff
@@ -0,0 +1,921 @@
+    const router = new Router('');
+    router.get({ path: '/', validate: false }, async (req, res) => res.ok({ content: 'ok' }));
+
+    const {
+      createCookieSessionStorageFactory,
+      registerAuth,
+      registerRouter,
+      server: innerServer,
+    } = await server.setup(config);
+    const sessionStorageFactory = await createCookieSessionStorageFactory<StorageData>(
+      cookieOptions
+    );
+    registerAuth((req, res, toolkit) => {
+      const user = { id: '42' };
+      const sessionStorage = sessionStorageFactory.asScoped(req);
+      sessionStorage.set({ value: user, expires: Date.now() + 1000 });
+      return toolkit.authenticated({ state: user });
+    });
+    registerRouter(router);
+    await server.start();
+
+    const response = await supertest(innerServer.listener)
+      .get('/')
+      .expect(200, { content: 'ok' });
+
+    expect(response.header['set-cookie']).toBeDefined();
+    const cookies = response.header['set-cookie'];
+    expect(cookies).toHaveLength(1);
+
+    const sessionCookie = request.cookie(cookies[0]);
+    if (!sessionCookie) {
+      throw new Error('session cookie expected to be defined');
+    }
+    expect(sessionCookie).toBeDefined();
+    expect(sessionCookie.key).toBe('sid');
+    expect(sessionCookie.value).toBeDefined();
+    expect(sessionCookie.path).toBe('/');
+    expect(sessionCookie.httpOnly).toBe(true);
+  });
+
+  it('allows manipulating cookies from route handler', async () => {
+    const {
+      createCookieSessionStorageFactory,
+      registerAuth,
+      registerRouter,
+      server: innerServer,
+    } = await server.setup(config);
+    const sessionStorageFactory = await createCookieSessionStorageFactory<StorageData>(
+      cookieOptions
+    );
+    registerAuth((req, res, toolkit) => {
+      const user = { id: '42' };
+      const sessionStorage = sessionStorageFactory.asScoped(req);
+      sessionStorage.set({ value: user, expires: Date.now() + 1000 });
+      return toolkit.authenticated();
+    });
+
+    const router = new Router('');
+    router.get({ path: '/', validate: false }, (req, res) => res.ok({ content: 'ok' }));
+    router.get({ path: '/with-cookie', validate: false }, (req, res) => {
+      const sessionStorage = sessionStorageFactory.asScoped(req);
+      sessionStorage.clear();
+      return res.ok({ content: 'ok' });
+    });
+    registerRouter(router);
+
+    await server.start();
+
+    const responseToSetCookie = await supertest(innerServer.listener)
+      .get('/')
+      .expect(200);
+
+    expect(responseToSetCookie.header['set-cookie']).toBeDefined();
+
+    const responseToResetCookie = await supertest(innerServer.listener)
+      .get('/with-cookie')
+      .expect(200);
+
+    expect(responseToResetCookie.header['set-cookie']).toEqual([
+      'sid=; Max-Age=0; Expires=Thu, 01 Jan 1970 00:00:00 GMT; HttpOnly; Path=/',
+    ]);
+  });
+
+  it.skip('is the only place with access to the authorization header', async () => {
```

Comment:
mean to keep this skipped?

Reply:
yes, I need to restore headers filtering to enable them again.